thoth/storages/graph/postgres.py

```diff
@@ -3775,6 +3775,112 @@ def create_python_cve_record(
 
                 return False
 
+    def update_missing_flag_package_version(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+        value: bool,
+    ) -> None:
+        """Update value of is_missing flag for PythonPackageVersion."""
+        with self._session_scope() as session:
+            subq = (session.query(PythonPackageVersion)
+                    .join(PythonPackageIndex)
+                    .filter(PythonPackageVersion.package_name == package_name)
+                    .filter(PythonPackageVersion.package_version == package_version)
+                    .filter(PythonPackageIndex.url == index_url)
+                    .with_entities(PythonPackageVersion.id)
+                    )
+            (
+                session.query(PythonPackageVersion)
+                .filter(PythonPackageVersion.id.in_(subq))
+                .update({"is_missing": value}, synchronize_session='fetch')
+            )
+
+    def get_adviser_run_origins_all(
+        self,
+        package_name: str = None,
+        package_version: str = None,
+        index_url: str = None,
+        start_offset: int = 0,
+        count: Optional[int] = DEFAULT_COUNT,
+        distinct: bool = False,
+    ) -> List[str]:
+        """Retrieve all origins (git repos URLs) in Adviser Run.
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_adviser_run_origins_all()
+        ['https://github.com/thoth-station/storages',
+         'https://github.com/thoth-station/user-api',
+         'https://github.com/thoth-station/adviser']
+        """
+        with self._session_scope() as session:
+            query = (
+                session.query(AdviserRun)
+                .order_by(AdviserRun.origin)
+                .order_by(AdviserRun.datetime.desc())
+            )
+
+            if package_name or package_version:
+                query = (
+                    query.join(
+                        ExternalPythonRequirementsLock,
+                        ExternalPythonRequirementsLock.python_software_stack_id == AdviserRun.user_sotware_stack_id
+                    )
+                    .join(PythonPackageVersion)
+                )
+                if index_url:
+                    query = query.join(PythonPackageIndex)
+
+            if package_name is not None:
+                package_name = self.normalize_python_package_name(package_name)
+                query.filter(PythonPackageVersion.package_name == package_name)
+
+            if package_version is not None:
+                package_version = self.normalize_python_package_version(package_version)
+                query.filter(PythonPackageVersion.package_version == package_version)
+
+            if index_url is not None:
```

Comment:
This should be also under `if package_name or package_version:` otherwise there would be a missing join.